### PR TITLE
GEODE-9217: Adding memory overhead tests for redis

### PR DESCRIPTION
**Pull request by upthewaterspout**
Adding integration tests that measure the memory overhead of both native redis
and radish.  These tests measure the memory used before and after adding
entries. They assert that the measured per entry overhead is within 1 byte of
the expected per entry overhead. They are measuring the all of the overhead -
whether it's from geode region entries or our radish data structures.

These are essentially the same tests from the feature/redis-memory-overhead
branch, but with assertions added that our expected overhead didn't change.

### CODE REVIEW

**Review by onichols-pivotal — APPROVED** [review]

File: gradle/test.gradle [hunk 1]
```diff
@@ -173,6 +173,12 @@ gradle.taskGraph.whenReady({ graph ->
         jvmArgs += ["--add-opens", "java.xml/jdk.xml.internal=ALL-UNNAMED"]
         jvmArgs += ["--add-opens", "java.base/jdk.internal.module=ALL-UNNAMED"]
         jvmArgs += ["--add-opens", "java.base/java.lang.module=ALL-UNNAMED"]
+        jvmArgs += ["--add-opens", "java.base/jdk.internal.ref=ALL-UNNAMED"]
+        jvmArgs += ["--add-opens", "java.base/jdk.internal.reflect=ALL-UNNAMED"]
+        jvmArgs += ["--add-opens", "java.base/jdk.internal.loader=ALL-UNNAMED"]
+        jvmArgs += ["--add-opens", "java.base/jdk.internal.misc=ALL-UNNAMED"]
+        jvmArgs += ["--add-opens", "java.base/jdk.internal.util.jar=ALL-UNNAMED"]
+        jvmArgs += ["--add-opens", "jdk.management/com.sun.management.internal=ALL-UNNAMED"]
```
Comment (onichols-pivotal) [inline gradle/test.gradle:181]: if these add-opens are needed for test, might they be needed elsewhere like dunit/internal/ProcessManager or session/tests/ServerContainer?

Comment (upthewaterspout) [reply gradle/test.gradle:181]: What's weird is each one of those things has a different set of add-opens than test.gradle, even before my changes. Should they be the same? In my case I just needed these for an integration test that is doing reflection to measure the memory usage of the cache.

**Review by nonbinaryprogrammer — APPROVED** [review]
looks good. we'll need to make sure that we add a test to this for sorted sets as well.

**Review by sabbey37 — APPROVED** [review]
Nice! 

Just wondering if we actually need/what the purpose of the `MemoryOverheadNativeRedisAcceptanceTest` is since we're not actually expecting our memory overhead values to be the same as Redis.

**Comment by upthewaterspout** [comment]
@sabbey37 - The main reason I left MemoryOverheadNativeRedisAcceptanceTest in there is so we can compare our numbers with native redis and see how close we are getting. It's also a framework and place to add new measurements - for example to measure the native redis overhead of sorted sets to compare with radish.

So I like having the code checked in somewhere. I guess I'm ambivalent about running it in CI every time. It takes about 4 seconds.

**Comment by sabbey37** [comment]
@upthewaterspout that makes sense and sounds reasonable!

File: gradle/test.gradle (hunk 1 above)
Comment (onichols-pivotal) [reply gradle/test.gradle:181]: ahh, so this is test code that needs the add-opens (not product code under test).  looks good then.

**Review by dschneider-pivotal — APPROVED** [review]

**Pull request merged**